cmd/setup-buildpack-build/main.go

```diff
@@ -72,6 +90,50 @@ func main() {
 	dockerutil.DescribeDefaultConfig(w)
 }
 
+// getBuildUser returns the UID and GID as specified by the buildpack builder
+// image environment variables subject to the platform build image spec:
+// https://github.com/buildpack/spec/blob/master/platform.md#build-image
+func getBuildUser(builderImage string) (uid, gid int, err error) {
+	imageRef, err := name.ParseReference(builderImage, name.WeakValidation)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	image, err := remote.Image(imageRef, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		return 0, 0, err
+	}
+
+	cfg, err := image.ConfigFile()
+	if err != nil {
+		return 0, 0, err
+	}
+
+	for _, val := range cfg.Config.Env {
+		parts := strings.Split(val, "=")
+		if len(parts) != 2 {
+			continue
+		}
+
+		key, value := parts[0], parts[1]
+		if key == "CNB_USER_ID" {
+			uid, err = strconv.Atoi(value)
+			if err != nil {
```

Comment:
Could you see a time where logging this error (and the similar CNB_GROUP_ID error) would be useful?

Reply:
the error should be logged up in the caller after it's returned from this function